### PR TITLE
tmuxifier: init at 0.13.0

### LINKED ISSUE #226393: Package request: tmuxifier
**Project description**

Tmuxifier allows you to easily create, edit, and load "layout" (template) files, which are simple shell scripts where you use the tmux command and helper commands provided by tmuxifier to manage Tmux sessions and windows.

Cool demo video: https://www.youtube.com/watch?v=ceRYL271cao

**Metadata**

* homepage URL: https://github.com/jimeh/tmuxifier
* source URL: https://github.com/jimeh/tmuxifier/tree/v0.13.0
* license: mit
* platforms: unix, linux, darwin,

### PR DESCRIPTION
**Pull request by wigust**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

This pull request will add requested in #226393 package. The Nix package tested on my machine and in a container builded with dockerTools.buildLayeredImage function.

fixes #226393
cc @adam248

### CODE REVIEW

File: pkgs/tools/misc/tmuxifier/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  version = "0.13.0";
+
+  src = fetchFromGitHub {
+    owner = "jimeh";
+    repo = "tmuxifier";
+    rev = "v${version}";
+    sha256 = "sha256-7TvJnvtZEo5h45PcSy3tJN09UblswV0mQbTaKjgLyqw=";
+  };
+
+  buildInputs = [ gnused ];
```
Comment (dotlambda) [inline pkgs/tools/misc/tmuxifier/default.nix:14]:
```suggestion
```

**Review by dotlambda — DISMISSED** [review]
`meta.maintainers` is missing

**Comment by dotlambda** [comment]
Please change the commit message to the PR title.

**Review by jian-lin — DISMISSED** [review]

**Comment by jian-lin** [comment]
> meta.maintainers is missing

I think it is optional.  Could you provide some pointers to the mandatory requriement?

**Comment by dotlambda** [comment]
> > meta.maintainers is missing
> 
> I think it is optional. Could you provide some pointers to the mandatory requriement?

It should be set: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -13894,6 +13894,8 @@ with pkgs;
 
   tmuxp = callPackage ../tools/misc/tmuxp { };
 
+  tmuxifier = callPackage ../tools/misc/tmuxifier { };
```
Comment (nagy) [inline pkgs/top-level/all-packages.nix:13897]: Could you use the new `by-name` structure introduced in #237439 ?

**Review by nagy — CHANGES_REQUESTED** [review]

File: pkgs/by-name/tm/tmuxifier/package.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    installShellCompletion --cmd tmuxifier --fish completion/tmuxifier.fish
+    installShellCompletion --cmd tmuxifier --zsh completion/tmuxifier.zsh
+  '';
+
+  meta = with lib; {
+    description = "Powerful session, window & pane management for Tmux";
+    homepage = "https://github.com/jimeh/tmuxifier";
+    license = licenses.mit;
+    maintainers = with maintainers; [ wigust ];
+    platforms = platforms.unix;
```
Comment (nagy) [inline pkgs/by-name/tm/tmuxifier/package.nix:39]: I forgot one thing, can you add `mainProgram = "tmuxifier";` here ?

**Review by nagy — CHANGES_REQUESTED** [review]

File: pkgs/by-name/tm/tmuxifier/package.nix [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    install -m 755 -d $out/bin/
+    install -m 755 -d $out/share/init
+
+    # Perform installation.
+    install -m 755 bin/tmuxifier $out/bin/
+    install -m 644 init.fish init.sh init.tcsh $out/share/init
+    cp -r lib libexec templates $out/share
+    installShellCompletion --cmd tmuxifier --bash completion/tmuxifier.bash
+    installShellCompletion --cmd tmuxifier --fish completion/tmuxifier.fish
+    installShellCompletion --cmd tmuxifier --zsh completion/tmuxifier.zsh
```
Comment (dotlambda) [inline pkgs/by-name/tm/tmuxifier/package.nix:31]: These three can be combined into one line.

File: pkgs/by-name/tm/tmuxifier/package.nix [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    sed -i "s@set -e@TMUXIFIER=$out/share\nTMUXIFIER_LAYOUT_PATH=\"\$HOME/.tmuxifier-layouts\"\nset -e@" bin/tmuxifier
+
+    # Create directories.
+    install -m 755 -d $out/bin/
+    install -m 755 -d $out/share/init
+
+    # Perform installation.
+    install -m 755 bin/tmuxifier $out/bin/
+    install -m 644 init.fish init.sh init.tcsh $out/share/init
+    cp -r lib libexec templates $out/share
```
Comment (dotlambda) [inline pkgs/by-name/tm/tmuxifier/package.nix:28]: We should probably use $out/lib and $out/libexec

File: maintainers/maintainer-list.nix [hunk 6]
```diff
@@ -18444,6 +18444,16 @@
     githubId = 7121530;
     name = "Wolf Honoré";
   };
+  wigust = {
+    name = "Oleg Pykhalov";
+    email = "go.wigust@gmail.com";
+    github = "wigust";
+    githubId = 7709598;
+    keys = [{
+      # primary: "C955 CC5D C048 7FB1 7966  40A9 199A F6A3 67E9 4ABB"
```
[the remark this answers is not in the document]
Comment (wigust) [reply maintainers/maintainer-list.nix:18453]: I know, but is it required for the primary key?

I don't see an example of this case in maintainers/maintainer-list.nix with publicly available key.

In Guix, which automatically verifies signed commits, primary keys are commented near subkey, where subkey was used for the signing commit.

File: pkgs/by-name/tm/tmuxifier/package.nix (hunk 5 above)
Comment (wigust) [reply pkgs/by-name/tm/tmuxifier/package.nix:28]: I tried this, but it breaks the tmuxifier if you install all or at least one $out/lib and $out/libexec

```
~/src/github.com/NixOS/nixpkgs$ result/bin/tmuxifier s mysession
result/bin/tmuxifier: line 34: /nix/store/qqbv9pjp2839rlzgrna1k7r8iv1ylb11-tmuxifier-0.13.0/share/lib/env.sh: No such file or directory

~/src/github.com/NixOS/nixpkgs$ result/bin/tmuxifier s mysession
result/bin/tmuxifier: line 41: tmuxifier-tmux-version: command not found
result/bin/tmuxifier: line 59: tmuxifier-resolve-command-path: command not found
tmuxifier: no such command 's'
```

**Review by nagy — APPROVED** [review]

File: pkgs/by-name/tm/tmuxifier/package.nix (hunk 5 above)
Comment (dotlambda) [reply pkgs/by-name/tm/tmuxifier/package.nix:25]: Then we should patch tmuxifier to use the correct path.

**Comment by adam248** [comment]
Thank you all for this. :heart:

**Comment by wigust** [comment]
@dotlambda all your suggestions are applied. Please, let me know if something is missing.

**Comment by wigust** [comment]
@jian-lin @dotlambda just a friendly ping. Could you mark your requested changes as resolved, please? (all your requested changes are applied)

**Comment by dotlambda** [comment]
I don't feel comfortable merging this because of the non-standard install phase. Someone should test if this actually works.

**Comment by wigust** [comment]
I use this package heavily in a container (builded with dockerTools.buildLayeredImage).  Also, I tested that ./result/bin/tmuxifier (create session and window, starting session) works as expected.

**Comment by wigust** [comment]
I don't fill that this patch will ever be merged, because there are 4 peoples in the review and I don't see if anybody would like to test it.

@dotlambda Is it possible to rewrite or drop things you don't like in install phase, so it could be finally merged?

It's kinda strange for me that the proposed thing is now blocking from the merge. If we choose rewriting, how the install phase should look like? Do we need *.patch files instead of 'sed' command invocation?

**Review by marsam — APPROVED** [review]

**Pull request merged**